src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMethodSymbolComparer.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Acuminator.Utilities.Common;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXOverrideMismatch
+{
+	public class PXOverrideMethodSymbolComparer : IEqualityComparer<IMethodSymbol>
+	{
+		public static PXOverrideMethodSymbolComparer Instance = new PXOverrideMethodSymbolComparer();
+
+		public bool Equals(IMethodSymbol src, IMethodSymbol ext)
+		{
+			var srcParams = src.Parameters.Select(p => p.Type).ToImmutableArray();
+			var extParams = ext.Parameters.Select(p => p.Type).ToImmutableArray();
+
+			var isSimple = srcParams.Length == extParams.Length;
+			var isComplex = srcParams.Length + 1 == extParams.Length;
+
+			if (!isSimple && !isComplex)
+			{
+				return false;
+			}
+
+			if (!HasCorrectOverrideSignature(src) || !HasCorrectAccessibility(src))
```

Comment:
The original method could be `protected` but the  pxoverride method must be public.

Reply:
You can override in a graph extension not generic public virtual or not generic protected virtual methods of the graph/lower level chained extensions. Thus,  you need to check that the original method is virtual and that both methods do not introduce their own generic type parameters.

Reply:
Both checks are done now.